### DISCUSSION TITLE
Automatic conversion of ANTLR4 grammar to Pest

### DISCUSSION THREAD

**Discussion by werediver** [pest-parser/pest#735] [Q&A] [answered]
The "X" part of my "XY problem" is that I want to parse Dart language with Pest. [Dart grammar](https://spec.dart.dev/DartLangSpecDraft.pdf) is huge, though. There is a [machine-readable version for ANTLR4](https://github.com/dart-lang/sdk/blob/320780856722112b645f6627b764793c4e74e976/tools/spec_parser/Dart.g) and I wonder if it would be easier to implement automatic ANTLR4-to-Pest conversion as a part of this endeavor.

I understand that ANTLR4 has features that are not supported by Pest. Hopefully, they are not used in the the above mentioned ANTLR4 Dart  grammar.

Are you aware of similar efforts? Do you see any caveats in ANTLR4-to-Pest conversion that I should be aware of?

**Comment by tomtau** [comment, marked as answer]
I think it won't be possible to do the full _ANTLR4-to-Pest_ conversion, but more likely a _subset-of-ANTLR4-to-Pest_ due to their differences.
I'm not aware of such efforts (but happy to learn if there are such efforts).

The closest efforts I could find:
* https://github.com/jarble/transpiler#generating-parsers-with-universal-transpiler -- the universal transpiler can support ANTLR and some PEG parser generators (like Peg.js), so it may be a good starting point to try to add Pest there?
* https://github.com/kaby76/Domemtech.Trash -- this ANTLR toolkit seems to support Pest conversion, but I guess it's only the opposite direction (_Pest-to-ANTLR4_), but I haven't looked into it in detail...
